src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResponseResource.java

```diff
@@ -0,0 +1,116 @@
+        this.complaintResponseRepository = complaintResponseRepository;
+        this.userService = userService;
+    }
+
+    /**
+     * POST /complaint-responses: create a new complaint response
+     *
+     * @param complaintResponse the complaint response to create
+     * @return the ResponseEntity with status 201 (Created) and with body the new complaint response
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/complaint-responses")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ComplaintResponse> createComplaint(@RequestBody ComplaintResponse complaintResponse) throws URISyntaxException {
+        log.debug("REST request to save ComplaintResponse: {}", complaintResponse);
+        if (complaintResponse.getId() != null) {
+            throw new BadRequestAlertException("A new complaint response cannot already have an id", ENTITY_NAME, "idexists");
+        }
+
+        if (complaintResponse.getComplaint() == null || complaintResponse.getComplaint().getId() == null) {
+            throw new BadRequestAlertException("A complaint response can be only associated to a complaint", ENTITY_NAME, "noresultid");
+        }
+
+        Long complaintId = complaintResponse.getComplaint().getId();
+        User reviewer = this.userService.getUser();
+
+        // Do not trust user input
+        Optional<Complaint> originalComplaintOptional = complaintRepository.findById(complaintId);
+
+        if (!originalComplaintOptional.isPresent()) {
+            throw new BadRequestAlertException("The complaint you are referring to does not exist", ENTITY_NAME, "noresult");
+        }
+
+        Complaint originalComplaint = originalComplaintOptional.get();
+        originalComplaint.setAccepted(true);
+
+        complaintResponse.setSubmittedTime(ZonedDateTime.now());
+        complaintResponse.setReviewer(reviewer);
+
+        complaintRepository.save(originalComplaint);
+
+        ComplaintResponse savedComplaint = complaintResponseRepository.save(complaintResponse);
+        return ResponseEntity.created(new URI("/api/complaint-responses/" + savedComplaint.getId()))
+                .headers(HeaderUtil.createEntityCreationAlert(ENTITY_NAME, savedComplaint.getId().toString())).body(savedComplaint);
+    }
+
+    /**
+     * GET /complaint-responses/:id : get the "id" complaint response.
+     *
+     * @param id the id of the complaint response to retrieve
+     * @return the ResponseEntity with status 200 (OK) and with body the complaint response, or with status 404 (Not Found)
+     */
+    @GetMapping("/complaint-responses/{id}")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ComplaintResponse> getComplaintResponse(@PathVariable Long id) {
+        log.debug("REST request to get ComplaintResponse : {}", id);
+        Optional<ComplaintResponse> complaintResponse = complaintResponseRepository.findById(id);
```

Comment:
we need a check that the user is allowed to see this

Reply:
Done